File: python/calculate_mode.py
def calculate_mode(li):
    this_dict = {}
    for i, x in enumerate(li):
        if x in this_dict:
            this_dict[x] += 1
        else:
            this_dict[x] = 1

    compare = 0
    for x in this_dict:
        if this_dict[x] > compare:
            compare = this_dict[x]
    print(compare)

    result = []
    for x in this_dict:
        if this_dict[x] == compare:
            result.append(x)

    return result
print(calculate_mode([1,2,3,4,5,4,4,5]))
print(calculate_mode([1,1,2,2,5,4,4,5]))
print(calculate_mode(["who", "what", "where", "who"]))

File: js/calculateMode.js
const calculateMode = (arr) => {
let countObj = {}
    for(let i = 0; i < arr.length; i++) {
        if(!countObj[arr[i]]){
            countObj[arr[i]] = 1
        } else {
            countObj[arr[i]] += 1
        }
    }
    //find largest duplicate and push value to compare
    let compare = 0
    for(let key in countObj) {
        if(compare < countObj[key]) {compare = countObj[key]}
    }
    //let result hold the values at compare index and return as Number if number and string if string
    let result = []
    for(let key in countObj) {
        if(compare == countObj[key] && isNaN(key) === true){result.push(key)}
        if(compare == countObj[key] && isNaN(Number(key)) === false){result.push(parseInt(key))}
    }
    
    return result
}

console.log(calculateMode([1,2,3,4,5,4,4,5])) //4
console.log(calculateMode([1,1,2,2,5,4,4,5])) //1,2,5,4
console.log(calculateMode(["who", "what", "where", "who"])) //who